mmgen/models/architectures/sngan_proj/modules.py

```diff
@@ -0,0 +1,445 @@
+
+    Args:
+        in_channels (int): Number of the channels of the input feature map.
+        num_classes (int): Number of the classes in the dataset. If ``use_cbn``
+            is True, ``num_classes`` must larger than 0.
+        use_cbn (bool, optional): Whether use conditional normalization. If
+            ``use_cbn`` is True, two embedding layers would be used to mapping
+            label to weight and bias used in normalization process.
+        norm_cfg (dict, optional): Config for normalization method. Default
+            to ``dict(type='BN')``.
+        auto_sync_bn (bool, optional): Whether convert Batch Norm to
+            Synchronized ones when Distributed training is on. Defualt to True.
+        norm_eps (float, optional): eps for Normalization layers (both conditional
+            and non-conditional ones). Default to 1e-4.
+        style (string, optional): Behavior and initialization style of the
+            module.  Default to 'BigGAN'
+    """
+
+    def __init__(self,
+                 in_channels,
+                 num_classes,
+                 use_cbn=True,
+                 norm_cfg=dict(type='BN'),
+                 cbn_norm_affine=False,
+                 auto_sync_bn=True,
+                 norm_eps=1e-4,
+                 style='BigGAN'):
+        super().__init__()
+        self.use_cbn = use_cbn
+        self.style = style
+
+        norm_cfg = deepcopy(norm_cfg)
+        norm_type = norm_cfg['type']
+
+        if norm_type not in ['IN', 'BN', 'SyncBN']:
+            raise ValueError('Only support `IN` (InstanceNorm), '
+                             '`BN` (BatcnNorm) and `SyncBN` for '
+                             'Class-conditional bn. '
+                             f'Receive norm_type: {norm_type}')
+
+        if self.use_cbn:
+            norm_cfg.setdefault('affine', cbn_norm_affine)
+        norm_cfg.setdefault('eps', norm_eps)
+        if check_dist_init() and auto_sync_bn and norm_type == 'BN':
+            norm_cfg['type'] = 'SyncBN'
+
+        _, self.norm = build_norm_layer(norm_cfg, in_channels)
+
+        if self.use_cbn:
+            if num_classes <= 0:
+                raise ValueError('`num_classes` must be larger '
+                                 'than 0 with `use_cbn=True`')
+            self.weight_embedding = nn.Embedding(num_classes, in_channels)
+            self.bias_embedding = nn.Embedding(num_classes, in_channels)
+
+        self.init_weights()
+
+    def forward(self, x, y=None):
+        """Forward function.
+
+        Args:
+            x (Tensor): Input tensor with shape (n, c, h, w).
+            y (Tensor, optional): Input label with shape (n, ).
+                Default None.
+
+        Returns:
+            Tensor: Forward results.
+        """
+
+        out = self.norm(x)
+        if self.use_cbn:
+            weight = self.weight_embedding(y)[:, :, None, None]
+            bias = self.bias_embedding(y)[:, :, None, None]
+            if self.style == 'BigGAN':
```

Comment:
Here, I may suggest using the original `reweight_embedding`. You just need to specify why we need this trick in comments.